apps/nrm/behaviours/save-and-exit.js

```diff
@@ -0,0 +1,57 @@
+'use strict';
+const saveAndExitService = require('../services/save-and-exit-service');
+
+/**
+ * Save data to database
+ *
+ * @param {object} data - request object
+ *
+ * @returns {Promise}
+ */
+const saveData = async(data) => {
+  return await saveAndExitService.write(data);
+};
+
+
+/**
+ * Get Session data
+ *
+ * @param {object} req - request object
+ *
+ * @returns {object} - the session data in the request object
+ */
+const getSessionData = (req) => {
+  const sessionData = req.sessionModel.toJSON();
+
+  return sessionData;
+};
+
+/**
+ * Save application progress
+ *
+ * Get the application data from the session then persist the data
+ * within a database
+ *
+ * @param {object} req - request object
+ *
+ * @returns {Promise} - id returned from saved application
+ */
+const saveApplication = async(req) => {
+  const sessionData = getSessionData(req);
+  return await saveData(sessionData);
+};
+
+module.exports = superclass => class extends superclass {
+  async saveValues(req, res) {
+    try {
+      const applicationId = await saveApplication(req);
+      // save resulting id to session
+      req.sessionModel.set('application-id', applicationId);
+
+      return res.redirect('/nrm/application-saved');
```

Comment:
These pages need to be added in a separate PR @sulthan-ahmed

Reply:
I think it's fine so long as it doesn't fail.  You could just have a route for it in HOF even if it's a blank page

Reply:
cool cheers, I think I'll do that.
I'll ensure this code follow only the success flow when inserting data to db, if any errors then show alt page.

Reply:
@sulthan-ahmed for the routing how can we determine whether the save and exit button was clicked?

Reply:
maybe something in the params

Reply:
hmm interesting question, I'd probably make the save button a link and use url parameter to determine that they've clicked on the link.  I think that's why it's a link on other services with save and return.  I don't think HOF will capture it in the session

Reply:
I think that's what Joe Fitter did with this https://github.com/UKHomeOfficeForms/hof-behaviour-address-lookup
see the url here
https://www.ending-a-tenancy.homeoffice.gov.uk/property-address?step=manual

Reply:
Cheers, That's what I eventually implemented.
When they click on the link they are redirected and I then save the values in getValues() method.

I didn't set the params however. As I wanted to get the response back before determining the page content

Reply:
Ahh ok, that makes sense.  I guess you don't need to set the params then